layers/cmd/ingest_csv.go

```diff
@@ -30,6 +31,86 @@ import (
 	"github.com/cloudprivacylabs/opencypher/graph"
 )
 
+type CSVIngester struct {
+	BaseIngestParams
+	StartRow  int
+	EndRow    int
+	HeaderRow int
+	ID        string
+}
+
+func (ci CSVIngester) Run(pipeline *PipelineContext) error {
+	layer, err := LoadSchemaFromFileOrRepo(pipeline.Context, ci.CompiledSchema, ci.Repo, ci.Schema, ci.Type, ci.Bundle)
+	if err != nil {
+		return err
+	}
+
+	reader := csv.NewReader(strings.NewReader(pipeline.InputFiles[0]))
+	startRow := ci.StartRow
+	endRow := ci.EndRow
+	headerRow := ci.HeaderRow
+	if headerRow >= startRow {
+		return errors.New("Header row is ahead of start row")
+
+	}
+	grph := pipeline.Graph
+	embedSchemaNodes := ci.EmbedSchemaNodes
+	onlySchemaAttributes := ci.OnlySchemaAttributes
+
+	parser := csvingest.Parser{
+		OnlySchemaAttributes: onlySchemaAttributes,
+		SchemaNode:           layer.GetSchemaRootNode(),
+	}
+
+	builder := ls.NewGraphBuilder(grph, ls.GraphBuilderOptions{
+		EmbedSchemaNodes:     embedSchemaNodes,
+		OnlySchemaAttributes: onlySchemaAttributes,
+	})
+
+	idTemplate := ci.ID
+	idTmp, err := template.New("id").Parse(idTemplate)
+	if err != nil {
+		return err
+	}
+	for row := 0; ; row++ {
+		rowData, err := reader.Read()
```

Comment:
Create a new graph for every row. that is, create a new builder in the for-loop. Ignore initial graph for csv